Fix Renew Cache Read Multi Bug

KTSCode opened the pull request:
## Bug:
When the context `{renew_cache: true}` is passed to a cached query in an application that is using a `cache_store` that supports `read_multi` (e.g. Redis) the cache is not cleared and `{renew_cache: true}` is ignored.

### Expected Behavior:
When the context `{renew_cache: true}` is passed to a cached query the cache is renewed.

## Description:
This commit fixes the above issue by adding logic that checks for `{renew_cache: true}` in the context of fragments passed to the `read_multi` method and returns `nil` for their value so that cache is refreshed. With this change, there is feature parity between cache stores that support `read_multi` and those that don't.

KTSCode commented:
@DmitryTsepelev 

I found this bug after trying everything to get the renew_cache functionality to work in a rails application that is using Redis for caching. 

Please let me know if there is anything else you want from me in order to get his PR merged. I'm happy to update the Changelog. I'm unsure if this change would push the gem version to 1.13.1 or 1.14.0. It should be compatible with anyone using the gem as described in the README, but will cause breaking changes if a user is expecting `{renew_cache: true}` to not work for caches that implement `read_multi`

DmitryTsepelev commented on lib/graphql/fragment_cache/fragment.rb:
```diff
@@ -12,22 +12,22 @@ class Fragment
-          unless FragmentCache.cache_store.respond_to?(:read_multi)
-            return fragments.map { |f| [f, f.read] }.to_h
-          end
+          return fragments.map { |f| [f, f.read] }.to_h unless FragmentCache.cache_store.respond_to?(:read_multi)
```
This is just a formatting change, right?

KTSCode replied:
Yes, it still functions exactly the same, the change is just from a later version of Rubocop's auto-formatting rules. I have format-on-save turned on in my text editor. I'll revert those changes so that only the code I changed and tested is modified.

DmitryTsepelev commented on lib/graphql/fragment_cache/fragment.rb:
```diff
@@ -12,22 +12,22 @@ class Fragment
 
-          fragments_to_cache_keys = fragments
-            .map { |f| [f, f.cache_key] }.to_h
+          fragments_to_cache_keys = fragments.map { |f| [f, f.cache_key] }.to_h
```
another format change

DmitryTsepelev commented on lib/graphql/fragment_cache/fragment.rb:
```diff
@@ -12,22 +12,22 @@ class Fragment
 
-          cache_keys_to_values = FragmentCache.cache_store.read_multi(*cache_keys)
+          # If there are cache_keys look up values with read_multi otherwise return an empty hash
+          cache_keys_to_values = unless cache_keys.empty?
```
I'd rather revert the condition since we have an else clause here 🙂

DmitryTsepelev reviewed (commented):
LGTM! Before we merge it in:

1. there are a couple of lines where the only change was in the formatting, I'd like to keep them as is (because it will make commits more readable)
2. Rubocop check is down, maybe because of the changes above

KTSCode commented:
Okay, I've reverted the formatting changes to lines that I didn't modify for the fix. I also installed the version of Rubocop and Stanard that are being used in CI and ran the Rubocop auto-formatter on both files I changed, so it should pass CI now.

DmitryTsepelev reviewed (approved):
LGTM! Thank you 🙂
